tests/inputs/example_service/test_example_service.py

```diff
@@ -0,0 +1,108 @@
+import asyncio
+from typing import AsyncIterator, AsyncIterable
+
+from grpclib.client import Channel
+from grpclib.server import Server
+
+from tests.output_betterproto.example_service.example_service import (
+    ExampleServiceImplementation,
+    ExampleServiceStub,
+    ExampleRequest,
+    ExampleResponse,
+)
+
+
+class ExampleService(ExampleServiceImplementation):
+    async def example_unary_unary(
+        self, example_string: str, example_integer: int
+    ) -> "ExampleResponse":
+        return ExampleResponse(
+            example_string=example_string,
+            example_integer=example_integer,
+        )
+
+    async def example_unary_stream(
+        self, example_string: str, example_integer: int
+    ) -> AsyncIterator["ExampleResponse"]:
+        response = ExampleResponse(
+            example_string=example_string,
+            example_integer=example_integer,
+        )
+        yield response
+        yield response
+        yield response
+
+    async def example_stream_unary(
+        self, example_request_iterator: AsyncIterable["ExampleRequest"]
+    ) -> "ExampleResponse":
+        async for example_request in example_request_iterator:
+            return ExampleResponse(
+                example_string=example_request.example_string,
+                example_integer=example_request.example_integer,
+            )
+
+    async def example_stream_stream(
+        self, example_request_iterator: AsyncIterable["ExampleRequest"]
+    ) -> AsyncIterator["ExampleResponse"]:
+        async for example_request in example_request_iterator:
+            yield ExampleResponse(
+                example_string=example_request.example_string,
+                example_integer=example_request.example_integer,
+            )
+
+
+async def async_test_server_start():
+    host = "127.0.0.1"
+    port = 13337
+
+    test_string = "test string"
+    test_int = 42
+
+    # start server
+    server = Server([ExampleService()])
+    await server.start(host, port)
```

Comment:
Do you try use the testing facility of grpclib?
e.g. https://github.com/danielgtaylor/python-betterproto/blob/master/tests/grpc/test_grpclib_client.py#L63
I think this is generally preferable to opening a socket etc.

Reply:
Also didn't know that was a thing. This makes the code a lot cleaner.